### ISSUE TITLE
Multiple host in connectionstring leads to queues being created on more than one host

### ISSUE THREAD

**Issue by janovesk** [Particular/NServiceBus.RabbitMQ#87]
Starting up a RabbitMQ endpoint with several hosts in the connection string and PurgeOnStartup(true) leads to some queues being created on the first host, and others on the later hosts.

Easily reproduced with SimpleRabbitMQ sample with multiplehosts in connectionstring and busConfiguration.PurgeOnStartup(true);

**Comment by janovesk** [comment]
Orginally posted here: https://groups.google.com/forum/#!topic/particularsoftware/NsMscrWibvs

**Comment by janovesk** [comment]
Ok, so the problem is that https://github.com/Particular/NServiceBus.RabbitMQ/blob/develop/src/NServiceBus.RabbitMQ/PersistentConnection.cs has the following code: 
```
void TryToConnect(object timer)
        {
            if (timer != null)
            {
                ((Timer) timer).Dispose();
            }

            Logger.Debug("Trying to connect");
            if (disposed)
            {
                return;
            }

            connectionFactory.Reset();
            do
            {
                try
                {
                    connection = connectionFactory.CreateConnection(purpose);
                    connectionFactory.Success();
                }
                catch (System.Net.Sockets.SocketException socketException)
                {
                    LogException(socketException);
                }
                catch (BrokerUnreachableException brokerUnreachableException)
                {
                    LogException(brokerUnreachableException);
                }
            } while (connectionFactory.Next());

            if (connectionFactory.Succeeded)
            {
                connection.ConnectionShutdown += OnConnectionShutdown;

                Logger.InfoFormat("Connected to RabbitMQ. Broker: '{0}', Port: {1}, VHost: '{2}'",
                                  connectionFactory.CurrentHost.Host,
                                  connectionFactory.CurrentHost.Port,
                                  connectionFactory.Configuration.VirtualHost);
            }
            else
            {
                Logger.ErrorFormat("Failed to connected to any Broker. Retrying in {0}", retryDelay);
                StartTryToConnect();
            }
        }
```
For every new connection attempt against RabbitMQ there is a ```connectionFactory.Reset();```. This invalidates the node currently selected and selects the next node.

**Comment by janovesk** [comment]
We have a couple of options here:
1. Clarify in documentation that this feature was intended for RabbitMQ clusters only, not multiple separate RabbitMQ instances. 
2. Disable auto-creation of queues when we detect multiple hosts in the connection string.
3. Disable auto-creation of queues when we detect multiple hosts in the connection string AND we notice that the hosts are not running in the same cluster(unsure on the technical feasability on this)
4. Update the implementation to only create auto-create queues on the first host of a multi-host connection string.
5. Drop support for multiple hosts in the connection string.

Any of the options should also update the documentation to stress that this feature was never intended as a load balancing solution, but as a fail over mechanism. We should document(or point to some documentation) on how a RabbitMQ cluster works and why you should put a load balancer in front of it to make sure you hit the active node. 

Thoughs @andreasohlund ?

**Comment by janovesk** [comment]
From https://www.rabbitmq.com/ha.html:

"A client can connect as normal to any node within a cluster. If that node should fail, and the rest of the cluster survives, then the client should notice the closed connection, and should be able to reconnect to some surviving member of the cluster. **Generally, it's not advisable to bake in node hostnames or IP addresses into client applications: this introduces inflexibility and will require client applications to be edited, recompiled and redeployed should the configuration of the cluster change or the number of nodes in the cluster change. Instead, we recommend a more abstracted approach: this could be a dynamic DNS service which has a very short TTL configuration, or a plain TCP load balancer, or some sort of mobile IP achieved with pacemaker or similar technologies.** In general, this aspect of managing the connection to nodes within a cluster is beyond the scope of RabbitMQ itself, and we recommend the use of other technologies designed specifically to solve these problems."

**Comment by janovesk** [comment]
Good resources: 
* http://insidethecpu.com/2014/11/17/load-balancing-a-rabbitmq-cluster/
* https://groups.google.com/forum/#!topic/rabbitmq-users/vvWAymjDww4

**Comment by janovesk** [comment]
Going with "Disable auto-creation of queues when we detect multiple hosts in the connection string." after discussion with @andreasohlund.

**Comment by andreasohlund** [comment]
@janovesk what release should we target for this one?

**Comment by janovesk** [comment]
After giving this some more thought and getting more user(@j--wong) [input] (https://github.com/Particular/NServiceBus.RabbitMQ/issues/89)  I'm pretty sure the best option is to just deprecate the whole multiple host in connection string feature for RabbitMQ

The code is written with the underlying assumption that if there are multiple hosts in a connection string, these are all part of the same cluster. 

1) If they are part of the same cluster everything works, but we are leading users away from the recommended approach for deploying a Rabbit cluster is behind a load balancer (ref: https://www.rabbitmq.com/clustering.html)

2) If they are not part of the same cluster, we mess up on the creation of queues and the real world usage scenario is not very clear at all. I'm calling YAGNI on it.

I propose we scrap the multi host connectionstrings completely, @andreasohlund.

**Comment by andreasohlund** [comment]
Agreed, let's deprecate, assigned to 3.0.0

**Comment by bording** [comment]
This change was released in 3.0.0.

For more detail about this change, see the [announcement](https://groups.google.com/d/topic/particularsoftware/8EkMRN97hVU/discussion) and the related [documentation](http://docs.particular.net/nservicebus/rabbitmq/cluster).

**Comment by adamralph** [comment]
Changing status to withdrawn. The multi-broker feature was removed in 3.0 so it does not exist in that version, and we will not be fixing this in older versions.
